xbmc/video/dialogs/GUIDialogTeletext.cpp

```diff
@@ -79,6 +79,29 @@ bool CGUIDialogTeletext::OnMessage(CGUIMessage& message)
   return CGUIDialog::OnMessage(message);
 }
 
+void CGUIDialogTeletext::DoProcess(unsigned int currentTime, CDirtyRegionList &dirtyregions)
+{
+  if (m_pTxtTexture)
+  {
+    m_TextDecoder.RenderPage();
```

Comment:
Since NeedRendering() is erased in Render via RenderingDone() the MarkDirtyRegion is not triggered correctly when switching teletext pages. Then we need a class flag indicating that something happened which is then evaluated in DoProcess which would be delayed by one frame always.